src/daft-core/src/array/ops/utf8.rs

```diff
@@ -603,6 +603,72 @@ impl Utf8Array {
         Ok(result)
     }
 
+    pub fn like(&self, pattern: &Utf8Array) -> DaftResult<BooleanArray> {
+        let (is_full_null, expected_size) = parse_inputs(self, &[pattern])
+            .map_err(|e| DaftError::ValueError(format!("Error in like: {e}")))?;
+        if is_full_null {
+            return Ok(BooleanArray::full_null(
+                self.name(),
+                &DataType::Boolean,
+                expected_size,
+            ));
+        }
+        if expected_size == 0 {
+            return Ok(BooleanArray::empty(self.name(), &DataType::Boolean));
+        }
+
+        let self_iter = create_broadcasted_str_iter(self, expected_size);
+        let pattern_iter = create_broadcasted_str_iter(pattern, expected_size);
+        let arrow_result = self_iter
+            .zip(pattern_iter)
+            .map(|(val, pat)| match (val, pat) {
+                (Some(val), Some(pat)) => {
+                    let pat = pat.replace('%', ".*").replace('_', ".");
+                    let re = regex::Regex::new(&format!("^{}$", pat));
```

Comment:
We probably don't want to create the same `Regex` object multiple times in a row if the pattern is the same.
For example if we have the query `df.filter(df["x"].like("pattern%"))` we shouldn't need to broadcast the pattern and should just create the Regex once!